packages/boa/esm/loader.mjs

```diff
@@ -37,3 +43,25 @@ export function dynamicInstantiate(url) {
     }
   };
 }
+
+// alternative dynamic loader
+export async function getSource(url, context, defaultGetSource) {
+  if (url.startsWith(protocol)) {
+    const moduleInstance = boa.import(url.replace(protocol, ''));
+    const moduleExports = Array.from(dir(moduleInstance) || []);
+    return {
+      source: `
+import boa from '../lib/index.js';
+
+const moduleInstance = boa.import('${url.replace(protocol, '')}');
+
+${moduleExports.map(name => `const __boa__${name} = moduleInstance['${name}'];`).join('\n')}
+
+export {
+${moduleExports.map(name => `  __boa__${name} as ${name},`).join('\n')}
```

Comment:
```suggestion
${moduleExports.map(name => `  ${name} = moduleInstance['${name}'],`).join('\n')}
```

Reply:
I think export might follow the syntax as:

```js
export {
  foo1 as foo2
}
```

I export in this way because:

- `foo1` should be a declared constant/variable.
- `__boa__` prefix is to prevent re-declaring built-in namespaces in javascript (e.g. `import`, `var`, `eval` which are likely to be used in python)

Reply:
Resolved.